### PR TITLE
feat(select): add clearable option to select multiple with test and english doc

### PR DESCRIPTION
**Pull request by wcastand**
## Checklist

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

Following #388 

Only missing piece is to update doc in `zh-cn`

### CODE REVIEW

File: components/select/select-multiple-value.tsx [hunk 1]
```diff
@@ -1,26 +1,33 @@
 import React from 'react'
 import useTheme from '../styles/use-theme'
 import Grid from '../grid'
+import SelectClearIcon from './select-icon-clear'
 
 interface Props {
   disabled: boolean
+  onClear: () => void | null
```
Comment (unix) [inline components/select/select-multiple-value.tsx:8]: got an error: `Type error: Type '(() => void) | null' is not assignable to type '() => void | null'.`

**Review by unix — COMMENTED** [review]
Great!! Thank you for your interest and continued support!
🎉 (Once all CIs are up and running, I will review and merge them in a timely manner.)

**Comment by unix** [comment]
By the way, the `features` all need to be merged into the `rc` branch.

**Comment by wcastand** [comment]
Should i reopen the PR to merge on RC then?

**Comment by wcastand** [comment]
sorry reflex i merged master again but fix conflict with rc on the last one :) 
hope it's okay

**Comment by unix** [comment]
@wcastand 
The history of PR looks a bit conflicted, perhaps you could try using `git rebase (to origin/rc)` to resolve these issues?

**Comment by wcastand** [comment]
not sure i'm doing this correctly.
on my repo since it's a fork, the branch called `remote` link to your repo so i did a `rebase remote/rc`
i had one conflict on package.json version which i fixed but then it was saying there was no change so i did a `rebase --skip` to finish the process.

If i do something bad, maybe i should look into starting over?

**Comment by unix** [comment]
@wcastand 
I've resolved the conflict and now it looks normal. (Our test coverage is down, maybe a little more optimization is needed here.)

**Comment by wcastand** [comment]
Added more test to the select, was missing a click handler on the clear icon.
Sorry again you had to help to fix my branch, always get confused with forks for some reason :(

**Review by unix — APPROVED** [review]
Perfect, looks like we're ready to merge this feature.

**Pull request merged**